plugins/src/gcc/imp.rs

```diff
@@ -0,0 +1,1274 @@
+
+            if pkt.departure >= self.group.departure {
+                if self.group.inter_departure_time_pkt(pkt) < *BURST_TIME {
+                    self.group.add(*pkt);
+                    continue;
+                }
+
+                if self.group.inter_arrival_time_pkt(pkt) < *BURST_TIME
+                    && self.group.inter_delay_variation_pkt(pkt) < Duration::zero()
+                {
+                    self.group.add(*pkt);
+                    continue;
+                }
+
+                let group = mem::take(&mut self.group);
+                gst::trace!(
+                    CAT,
+                    "Packet group done: {:?}",
+                    gst::ClockTime::from_nseconds(group.departure.num_nanoseconds().unwrap() as u64)
+                );
+                if let Some(prev_group) = mem::replace(&mut self.prev_group, Some(group.clone())) {
+                    self.last_inter_departure_times
+                        .push(group.inter_departure_time(&prev_group));
+                    if self.last_inter_departure_times.len() > LAST_INTER_DEPARTURE_SIZE {
+                        self.last_inter_departure_times.remove(0);
+                    }
+                    self.kalman_estimate(&prev_group, &group);
+                    self.overuse_filter();
+                }
+            } else {
+                gst::debug!(
+                    CAT,
+                    "Ignoring packet departed at {:?} as we got feedback too late",
+                    gst::ClockTime::from_nseconds(pkt.departure.num_nanoseconds().unwrap() as u64)
+                );
+            }
+        }
+
+        self.compute_loss_average(lost_packets as f64 / packets.len() as f64);
+    }
+
+    fn compute_loss_average(&mut self, loss_fraction: f64) {
+        let now = time::Instant::now();
+
+        if let Some(ref last_update) = self.last_loss_update {
+            self.loss_average = loss_fraction
+                + (-Duration::from_std(now - *last_update)
+                    .unwrap()
+                    .num_milliseconds() as f64)
+                    .exp()
+                    * (self.loss_average - loss_fraction);
+        }
+
+        self.last_loss_update = Some(now);
+    }
+
+    fn kalman_estimate(&mut self, prev_group: &PacketGroup, group: &PacketGroup) {
+        self.measure = group.inter_delay_variation(prev_group);
+
+        let z = self.measure - self.estimate;
+        let zms = z.num_microseconds().unwrap() as f64 / 1000.0;
+
+        // FIXME: This doesn't exactly follows the spec as we should compute and
+        // use f_max here.
+        let alpha = ONE_MINUS_CHI.powf(30.0 / (1000. * 5. * 1_000_000.));
+        let root = self.measurement_uncertainty.sqrt();
+        let root3 = 3. * root;
+
+        if zms > root3 {
+            self.measurement_uncertainty =
+                (alpha * self.measurement_uncertainty + (1. - alpha) * root3 * root3).max(1.);
+        }
+        self.measurement_uncertainty =
```

Comment:
So, if I read the RFC right:

> If z(i) > 3*sqrt(var_v_hat) the filter is updated with 3*sqrt(var_v_hat) rather than z(i).

Shouldn't this be in an else clause?